### ISSUE TITLE
Old pictures need to be deleted from the database

### ISSUE THREAD

**Issue by marqqs** [wintermeyer/animina#1093]
Pictures which aren't in use anymore are still accessible via deep link.
They should be deleted from the system instead.
AFAIK this might be necessary according to EU law.

**Comment by wintermeyer** [comment]
WOW! That is an embarrassing bug. 😱

Thank you @marqqs for finding and submitting that! We will take take of it.

**Comment by wintermeyer** [comment]
Just to be sure that we don't fall into the same trap a second time: This includes all versions of an image (e.g. thumbnails). When an image is deleted or updated ALL the old ones have to be deleted not just from the database but from the filesystem too.

Please add a test for this. I do want to be sure that the app takes care of that!

**Comment by Kimutai01** [comment]
@wintermeyer 
What about about me? If someone uploads a new about me photo do we need to delete it?

**Comment by wintermeyer** [comment]
> @wintermeyer What about about me? If someone uploads a new about me photo do we need to delete it?

@Kimutai01 Please think about it for a moment. Why would we want to not delete an about me photo if we delete all the others? Let me give you an example: Maybe I published a photo of myself and in the background you can see a post-it with my WiFi password. I realise my mistake right away and delete/update the photo. Of course I want animina to delete ALL old versions of that photo. There is no reason to keep those photos. The user deleted them. So we have to do the same on the filesystem.

**Comment by Kimutai01** [comment]
> > @wintermeyer What about about me? If someone uploads a new about me photo do we need to delete it?
> 
> 
> 
> @Kimutai01 Please think about it for a moment. Why would we want to not delete an about me photo if we delete all the others? Let me give you an example: Maybe I published a photo of myself and in the background you can see a post-it with my WiFi password. I realise my mistake right away and delete/update the photo. Of course I want animina to delete ALL old versions of that photo. There is no reason to keep those photos. The user deleted them. So we have to do the same on the filesystem.

It's okay let me work on it.

**Comment by marqqs** [comment]
> What about about me? If someone uploads a new about me photo do we need to delete it?

That's how I discovered this behavior. I still had an open browser tab with the link to the old about-me picture. Now, days after I updated the picture of the about-me story, the old picture is still accessible via deep link (the old and still open browser tab) although not used anywhere in the profile. I think this is not a browser cache thing – I forced relaod by pressing Ctrl-F5.
